### PR TITLE
Be able to set the hash within the request. Fix #17

### LINKED ISSUE #17: [Feature] Be able to set the hash
We would want to set the hash instead of using the one computed by insight (sha1(preview_url)). It should be optional for backward-compatibility.

### PR DESCRIPTION
**Pull request by Lothiraldan**

### CODE REVIEW

**Comment by Natim** [comment]
Are you sure you want to copy past the defaut hash in all the backends?

**Comment by Lothiraldan** [comment]
Right, we can refactor this part of the code.

File: insight_reloaded/api.py [hunk 1]
```diff
@@ -78,6 +78,11 @@ def get(self, queue):
         except ValueError:
             params['crop'] = CROP_SIZE
 
+        try:
+            params['hash'] = self.get_argument('hash')
+        except ValueError:
+            pass
```
Comment (Natim) [inline insight_reloaded/api.py]: lets say here that if the hash is missing, we defined it with `hashlib.sha1(self.document_url).hexdigest()`

File: insight_reloaded/worker.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def start_worker():
 
         # Create the storage
         storage_class = resolve_name(STORAGE_CLASS)
-        storage = storage_class(params['url'])
+        storage = storage_class(params['url'], params.get('hash'))
```
Comment (Natim) [inline insight_reloaded/worker.py]: params['hash']

File: insight_reloaded/storage/s3.py [hunk 3]
```diff
@@ -10,9 +10,11 @@
 
 class S3Storage(object):
 
-    def __init__(self, document_url, s3_location=None):
+    def __init__(self, document_url, document_hash=None, s3_location=None):
```
Comment (Natim) [inline insight_reloaded/storage/s3.py]: document_hash (can't be none)

**Comment by Natim** [comment]
Can you squash: http://tech.novapost.fr/successfull-git-feature-workflow-in-team.html

**Comment by Lothiraldan** [comment]
Good ?

**Pull request merged**